apps/bfd-pipeline/bfd-pipeline-rif-extract/src/test/java/gov/cms/bfd/pipeline/rif/extract/s3/DataSetMonitorIT.java

```diff
@@ -178,4 +183,126 @@ public void multipleDataSetsTest() throws InterruptedException {
       if (bucket != null) DataSetTestUtilities.deleteObjectsAndBucket(s3Client, bucket);
     }
   }
+
+  /**
+   * Tests {@link gov.cms.bfd.pipeline.rif.extract.s3.DataSetMonitor} when run against a bucket with
+   * two data sets in it that do not appear at once.
+   *
+   * @throws InterruptedException (shouldn't happen)
+   */
+  @Test
+  public void delayedMultipleDataSetsTest() throws InterruptedException {
+    ScheduledExecutorService manifestUploaderService = Executors.newScheduledThreadPool(3);
+
+    class ManifestUploader implements Runnable {
+      AmazonS3 s3Client;
+      Bucket bucket;
+      DataSetManifest manifest;
+      URL manifestResourceURL;
+
+      public ManifestUploader(
+          AmazonS3 s3Client, Bucket bucket, DataSetManifest manifest, URL manifestResourceURL) {
+        this.s3Client = s3Client;
+        this.bucket = bucket;
+        this.manifest = manifest;
+        this.manifestResourceURL = manifestResourceURL;
+      }
+
+      public void run() {
+        s3Client.putObject(DataSetTestUtilities.createPutRequest(bucket, manifest));
+        s3Client.putObject(
+            DataSetTestUtilities.createPutRequest(
+                bucket, manifest, manifest.getEntries().get(0), manifestResourceURL));
+      }
+    }
+
+    AmazonS3 s3Client = S3Utilities.createS3Client(new ExtractionOptions("foo"));
+    Bucket bucket = null;
+
+    try {
+      /*
+       * Create the (empty) bucket to run against, and populate it with
+       * two data sets.
+       */
+      bucket = DataSetTestUtilities.createTestBucket(s3Client);
+      ExtractionOptions options = new ExtractionOptions(bucket.getName());
+      LOGGER.info(
+          "Bucket created: '{}:{}'",
+          s3Client.getS3AccountOwner().getDisplayName(),
+          bucket.getName());
+
+      DataSetManifest manifestA =
+          new DataSetManifest(
+              Instant.now().minus(1L, ChronoUnit.HOURS),
+              0,
+              new DataSetManifestEntry("beneficiaries.rif", RifFileType.BENEFICIARY));
+      manifestUploaderService.schedule(
+          new ManifestUploader(
+              s3Client, bucket, manifestA, StaticRifResource.SAMPLE_A_BENES.getResourceUrl()),
+          new Random().nextInt(10),
+          TimeUnit.SECONDS);
+
+      DataSetManifest manifestB =
+          new DataSetManifest(
+              manifestA.getTimestampText(),
+              1,
+              new DataSetManifestEntry("pde.rif", RifFileType.PDE));
+      manifestUploaderService.schedule(
+          new ManifestUploader(
+              s3Client, bucket, manifestB, StaticRifResource.SAMPLE_A_PDE.getResourceUrl()),
+          new Random().nextInt(10),
```

Comment:
If I'm reading this right, this test case could take at least 20 seconds, depending on your randoms? That's **slow**, as we tend to consider these things.

Can you think through and document: what's the minimum amount of time this test needs to be wait in order to be valid? Then, maybe remove the randoms and just always wait exactly that long.